Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm

```diff
@@ -104,13 +104,17 @@ - (instancetype)initWithWebExtensionAction:(WebExtensionAction&)action
 
 - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler
 {
-    if (!_webExtensionAction || !_webExtensionAction->extensionContext()) {
+    if (!_webExtensionAction) {
         decisionHandler(WKNavigationActionPolicyCancel);
         return;
     }
 
+    RefPtr extensionContext = _webExtensionAction->extensionContext();
+    if (!extensionContext)
+        return;
```

Comment:
Need to call `decisionHandler` here too.
```
decisionHandler(WKNavigationActionPolicyCancel);
```

Reply:
Good point. Thanks for catching.